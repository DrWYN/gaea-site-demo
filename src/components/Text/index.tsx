import * as React from 'react';

interface ITextProps {
  testID?: string;
  className?: string;
  style?: any;
  numberOfLines?: number;
  children?: any
}

const Text = (props: ITextProps) => {
  const { style = {}, numberOfLines = 0, testID, ...other } = props;
  const lineStyle = numberOfLines > 0 ? {
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    wordBreak: 'break-word',
    WebkitLineClamp: numberOfLines,
    display: '-webkit-box',
    WebkitBoxOrient: 'vertical',
  } : {wordWrap: 'break-word'}
  
  return (
    <span data-testid={testID} {...other}
      style={{
        ...style,
        ...lineStyle
      }}/>
  ) 
}

export default Text;
