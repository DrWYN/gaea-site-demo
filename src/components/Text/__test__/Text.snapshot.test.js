import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../index';
describe('<Text />', () => {
  it('no props', () => {
    const component = renderer.create(<Text />);

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();

  });

  it('props testID', () => {
    const component = renderer.create(<Text testID="test"/>);

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  
  it('props className', () => {
    const component = renderer.create(<Text className="class"/>);

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  
  it('props numberOfLines', () => {
    const component = renderer.create(<Text numberOfLines={2}/>);

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  
  it('props children', () => {
    const component = renderer.create(<Text>aaa <Text>bbb</Text></Text>);

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
