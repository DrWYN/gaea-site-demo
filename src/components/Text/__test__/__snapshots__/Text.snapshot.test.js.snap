// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> no props 1`] = `
<span
  data-testid={undefined}
  style={
    Object {
      "wordWrap": "break-word",
    }
  }
/>
`;

exports[`<Text /> props children 1`] = `
<span
  data-testid={undefined}
  style={
    Object {
      "wordWrap": "break-word",
    }
  }
>
  aaa 
  <span
    data-testid={undefined}
    style={
      Object {
        "wordWrap": "break-word",
      }
    }
  >
    bbb
  </span>
</span>
`;

exports[`<Text /> props className 1`] = `
<span
  className="class"
  data-testid={undefined}
  style={
    Object {
      "wordWrap": "break-word",
    }
  }
/>
`;

exports[`<Text /> props numberOfLines 1`] = `
<span
  data-testid={undefined}
  style={
    Object {
      "WebkitBoxOrient": "vertical",
      "WebkitLineClamp": 2,
      "display": "-webkit-box",
      "overflow": "hidden",
      "textOverflow": "ellipsis",
      "wordBreak": "break-word",
    }
  }
/>
`;

exports[`<Text /> props testID 1`] = `
<span
  data-testid="test"
  style={
    Object {
      "wordWrap": "break-word",
    }
  }
/>
`;
